src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Expressions/ExpressionHelper.cs

```diff
@@ -58,7 +58,18 @@ public static string GetExpressionText([NotNull] LambdaExpression expression)
                 else if (part.NodeType == ExpressionType.MemberAccess)
                 {
                     var memberExpressionPart = (MemberExpression)part;
-                    nameParts.Push("." + memberExpressionPart.Member.Name);
+                    var name = memberExpressionPart.Member.Name;
+
+                    // If identifier contains "__", it is "reserved for use by the implementation" and likely compiler-
+                    // or Razor-generated e.g. the name of a field in a delegate's generated class.
+                    if (name.Contains("__"))
```

Comment:
Another way to verify this would probably to look at `memberExpressionPart.Member` and it's declaring class, which has `[CompilerGenerated]`. 

What we have here is probably good enough, but just in case we need to be more certain in the future.